cmd/tailscale/cli/up.go

```diff
@@ -881,13 +881,22 @@ func checkForAccidentalSettingReverts(newPrefs, curPrefs *ipn.Prefs, env upCheck
 	return errors.New(sb.String())
 }
 
-// applyImplicitPrefs mutates prefs to add implicit preferences. Currently
-// this is just the operator user, which only needs to be set if it doesn't
+// applyImplicitPrefs mutates prefs to add implicit preferences for the user operator.
+// If the operator flag is passed no action is taken, otherwise this only needs to be set if it doesn't
 // match the current user.
-//
-// curUser is os.Getenv("USER"). It's pulled out for testability.
-func applyImplicitPrefs(prefs, oldPrefs *ipn.Prefs, curUser string) {
-	if prefs.OperatorUser == "" && oldPrefs.OperatorUser == curUser {
+func applyImplicitPrefs(prefs, oldPrefs *ipn.Prefs, env upCheckEnv) {
+	explicitlySet := false
+	upFlagSet.Visit(func(f *flag.Flag) {
+		if f.Name == "operator" {
+			explicitlySet = true
+		}
+	})
+
+	if explicitlySet && prefs.OperatorUser == "" {
```

Comment:
merge this check into line 899, adding a `&& !explicitOperator`?

then we can add future implicit prefs without this short-circuiting the rest of the func.

(rename `explicitlySet` to `explicitOperator` too, to be clear which thing's explicit)

Reply:
happy to make the change. Is this for better readability or performance?

Reply:
This reverted.